src/main/java/org/cryptomator/windows/common/RegistryKey.java

```diff
@@ -0,0 +1,181 @@
+	public static final RegistryKey HKEY_USERS = new RegistryRoot(Winreg_h.HKEY_USERS(), "HKEY_USERS");
+
+	private final String path;
+	private MemorySegment handle;
+	private volatile boolean isClosed = false;
+
+	RegistryKey(MemorySegment handle, String path) {
+		this.handle = handle;
+		this.path = path;
+	}
+
+	//-- GetValue functions --
+
+	public String getStringValue(String name, boolean isExpandable) throws RegistryValueException {
+		try (var arena = Arena.ofConfined()) {
+			var data = getValue(arena, name, isExpandable ? RRF_RT_REG_EXPAND_SZ() : RRF_RT_REG_SZ());
+			return data.getString(0, StandardCharsets.UTF_16LE);
+		}
+	}
+
+	public int getDwordValue(String name) throws RegistryValueException {
+		try (var arena = Arena.ofConfined()) {
+			var data = getValue(arena, name, RRF_RT_REG_DWORD());
+			return data.get(ValueLayout.JAVA_INT, 0);
+		}
+	}
+
+	private MemorySegment getValue(Arena arena, String name, int dwFlags) throws RegistryValueException {
+		var lpValueName = arena.allocateFrom(name, StandardCharsets.UTF_16LE);
+		var lpDataSize = arena.allocateFrom(ValueLayout.JAVA_INT, 0);
+
+		int result;
+		int bufferSize = 128; //will be doubled in first iteration
+		MemorySegment lpData;
+		do {
+			bufferSize = bufferSize << 1;
+			if (bufferSize >= MAX_DATA_SIZE) {
+				throw new RuntimeException("Getting value %s for key %s failed. Maximum buffer size of %d reached.".formatted(lpValueName.getString(0, StandardCharsets.UTF_16LE), path, bufferSize));
+			}
+			lpData = arena.allocate(bufferSize);
+			lpDataSize.set(ValueLayout.JAVA_INT, 0, bufferSize);
+
+			result = Winreg_h.RegGetValueW(handle, NULL, lpValueName, dwFlags, NULL, lpData, lpDataSize);
+
+		} while (result == ERROR_MORE_DATA());
+
+		if (result == ERROR_SUCCESS()) {
+			return lpData;
+		} else {
+			throw new RegistryValueException("winreg_h:RegGetValue", path, lpValueName.getString(0, StandardCharsets.UTF_16LE), result);
+		}
+	}
+
+	//-- SetValue functions --
+
+	public void setStringValue(String name, String data, boolean isExpandable) throws RegistryValueException {
+		try (var arena = Arena.ofConfined()) {
+			var lpValueName = arena.allocateFrom(name, StandardCharsets.UTF_16LE);
+			var lpValueData = arena.allocateFrom(data, StandardCharsets.UTF_16LE);
+			setValue(lpValueName, lpValueData, isExpandable ? REG_EXPAND_SZ() : Winreg_h.REG_SZ());
+		}
+	}
+
+	public void setDwordValue(String name, int data) throws RegistryValueException {
+		try (var arena = Arena.ofConfined()) {
+			var lpValueName = arena.allocateFrom(name, StandardCharsets.UTF_16LE);
+			var lpValueData = arena.allocateFrom(ValueLayout.JAVA_INT, data);
+			setValue(lpValueName, lpValueData, Winreg_h.REG_DWORD());
+		}
+	}
+
+	private void setValue(MemorySegment lpValueName, MemorySegment data, int dwFlags) throws RegistryValueException {
+		if (data.byteSize() > MAX_DATA_SIZE) {
+			throw new IllegalArgumentException("Data must be smaller than " + MAX_DATA_SIZE + "bytes.");
+		}
```

Comment:
This would break API symmetry: We can set values larger than `MAX_DATA_SIZE`. But getting such values is not possible, because there we have the restriction `valueData =< MAX_DATA_SIZE`.

Reply:
Ok if this is intentional, I am fine with it.